doc/changelog/05-tactic-language/12254-wit-ref-dyn.rst

```diff
@@ -0,0 +1,5 @@
+- **Changed:**
+  The "reference" tactic generic argument now accepts arbitrary
+  variables in non-strict mode
+  (`#12254 <https://github.com/coq/coq/pull/12254>`_,
```

Comment:
If the point is to include local (goal) variables in `reference` (see below), I would suggest:
```suggestion
  The "reference" tactic generic argument now accepts arbitrary
   variables of the goal context
  (`#12254 <https://github.com/coq/coq/pull/12254>`_,
```
The fact that it accepts them in strict mode would then just derive from the general statement.

Reply:
Will do.